Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2007-06-12
Initial Package Version: 1.17
Upstream Status: Merged
Origin: Upstream
Description: Fixes test 12 (exclude.at)

diff -Naur tar-1.17.orig/ChangeLog tar-1.17/ChangeLog
--- tar-1.17.orig/ChangeLog	2007-06-08 10:33:57.000000000 +0100
+++ tar-1.17/ChangeLog	2007-06-12 13:11:18.000000000 +0100
@@ -1,3 +1,7 @@
+2007-06-09  Sergey Poznyakoff  <gray@gnu.org.ua>
+
+	* tests/exclude.at: Sort the output.
+
 2007-06-08  Sergey Poznyakoff  <gray@gnu.org.ua>
 
 	* configure.ac, NEWS: Raise version number to 1.17
diff -Naur tar-1.17.orig/tests/exclude.at tar-1.17/tests/exclude.at
--- tar-1.17.orig/tests/exclude.at	2007-01-23 10:25:57.000000000 +0000
+++ tar-1.17/tests/exclude.at	2007-06-12 13:09:08.000000000 +0100
@@ -24,6 +24,8 @@
 AT_KEYWORDS([exclude])
 
 AT_TAR_CHECK([
+AT_SORT_PREREQ
+
 mkdir dir
 echo blues > dir/blues
 echo jazz > dir/jazz
@@ -37,79 +39,79 @@
 for option in exclude-caches exclude-caches-under exclude-caches-all 
 do
   echo OPTION $option
-  tar -cf archive.tar --$option -v dir 2>err
+  tar -cf archive.tar --$option -v dir 2>err | sort
   cat err
   echo ARCHIVE
-  tar tf archive.tar
+  tar tf archive.tar | sort
 done
 
 for option in exclude-tag exclude-tag-under exclude-tag-all 
 do
   echo OPTION $option
-  tar -cf archive.tar --${option}=tagfile -v dir 2>err
+  tar -cf archive.tar --${option}=tagfile -v dir 2>err | sort
   cat err
   echo ARCHIVE
-  tar tf archive.tar
+  tar tf archive.tar | sort
 done
 ],
 [0],
 [OPTION exclude-caches
 dir/
 dir/blues
-dir/jazz
 dir/folk/
-dir/folk/tagfile
 dir/folk/sanjuan
+dir/folk/tagfile
+dir/jazz
 dir/rock/
 dir/rock/CACHEDIR.TAG
 tar: dir/rock/: contains a cache directory tag CACHEDIR.TAG; contents not dumped
 ARCHIVE
 dir/
 dir/blues
-dir/jazz
 dir/folk/
-dir/folk/tagfile
 dir/folk/sanjuan
+dir/folk/tagfile
+dir/jazz
 dir/rock/
 dir/rock/CACHEDIR.TAG
 OPTION exclude-caches-under
 dir/
 dir/blues
-dir/jazz
 dir/folk/
-dir/folk/tagfile
 dir/folk/sanjuan
+dir/folk/tagfile
+dir/jazz
 dir/rock/
 tar: dir/rock/: contains a cache directory tag CACHEDIR.TAG; contents not dumped
 ARCHIVE
 dir/
 dir/blues
-dir/jazz
 dir/folk/
-dir/folk/tagfile
 dir/folk/sanjuan
+dir/folk/tagfile
+dir/jazz
 dir/rock/
 OPTION exclude-caches-all
 dir/
 dir/blues
-dir/jazz
 dir/folk/
-dir/folk/tagfile
 dir/folk/sanjuan
+dir/folk/tagfile
+dir/jazz
 tar: dir/rock/: contains a cache directory tag CACHEDIR.TAG; directory not dumped
 ARCHIVE
 dir/
 dir/blues
-dir/jazz
 dir/folk/
-dir/folk/tagfile
 dir/folk/sanjuan
+dir/folk/tagfile
+dir/jazz
 OPTION exclude-tag
 dir/
 dir/blues
-dir/jazz
 dir/folk/
 dir/folk/tagfile
+dir/jazz
 dir/rock/
 dir/rock/CACHEDIR.TAG
 dir/rock/file
@@ -117,17 +119,17 @@
 ARCHIVE
 dir/
 dir/blues
-dir/jazz
 dir/folk/
 dir/folk/tagfile
+dir/jazz
 dir/rock/
 dir/rock/CACHEDIR.TAG
 dir/rock/file
 OPTION exclude-tag-under
 dir/
 dir/blues
-dir/jazz
 dir/folk/
+dir/jazz
 dir/rock/
 dir/rock/CACHEDIR.TAG
 dir/rock/file
@@ -135,8 +137,8 @@
 ARCHIVE
 dir/
 dir/blues
-dir/jazz
 dir/folk/
+dir/jazz
 dir/rock/
 dir/rock/CACHEDIR.TAG
 dir/rock/file
